examples/run_CPMG.py

```diff
@@ -4,149 +4,184 @@
-    data.reorder("t2", first=False)
-    for_plot = data.C
-    for_plot.ft('t2',shift=True)
-    for_plot.ft(['ph1'], unitary = True)
-    fl.next('FTed data')
-    fl.image(for_plot.C.mean("nScans")
-    )
-else:
-    if config_dict["nScans"] > 1:
-        data.setaxis("nScans", r_[0 : config_dict["nScans"]])
-    data.rename("t","t2")
-    fl.next("Raw - time")
-    fl.image(
-        data.C.mean("nScans"))
-    data.reorder("t2", first=False)
-    for_plot = echo_data.C
-    for_plot.ft('t2',shift=True)
-    fl.next('FTed data')
-    fl.image(for_plot)
-data.name(config_dict["type"] + "_" + config_dict["cpmg_counter"])
-data.set_prop("postproc_type", "spincore_CPMGv2")
+data.chunk(
+    "t",
+    ["ph2", "ph_diff", "nEcho", "t2"],
+    [len(ph2), len(ph_diff), int(config_dict["nEchoes"]), -1],
+)
+data.setaxis("nEcho", r_[0 : int(config_dict["nEchoes"])]).setaxis(
+    "ph2", ph2 / 4
+).setaxis("ph_diff", ph_diff / 4)
+data.set_prop("postproc_type", "spincore_diffph_SE_v1")
+data.set_prop("coherence_pathway", {"ph_overall": -1, "ph1": +1})
 data.set_prop("acq_params", config_dict.asdict())
-target_directory = getDATADIR(exp_type="ODNP_NMR_comp/CPMG")
+nodename = config_dict["type"] + "_" + str(config_dict["cpmg_counter"])
+data.name(nodename)
 filename_out = filename + ".h5"
-nodename = data.name()
 if os.path.exists(f"{filename_out}"):
     print("this file already exists so we will add a node to it!")
     with h5py.File(
         os.path.normpath(os.path.join(target_directory, f"{filename_out}"))
     ) as fp:
-        if nodename in fp.keys():
-            print("this nodename already exists, so I will call it temp_cpmg")
-            data.name("temp_cpmg")
-            nodename = "temp_cpmg"
-    data.hdf5_write(f"{filename_out}", directory=target_directory)
-else:
-    try:
-        data.hdf5_write(f"{filename_out}", directory=target_directory)
-    except:
-        print(
-            f"I had problems writing to the correct file {filename}.h5, so I'm going to try to save your file to temp_cpmg.h5 in the current h5 file"
-        )
-        if os.path.exists("temp_cpmg.h5"):
-            print("there is a temp_cpmg.h5 already! -- I'm removing it")
-            os.remove("temp_cpmg.h5")
-            data.hdf5_write("temp_cpmg.h5")
-            print(
-                "if I got this far, that probably worked -- be sure to move/rename temp_cpmg.h5 to the correct name!!"
-            )
+        tempcounter = 1
+        orig_nodename = nodename
+        while nodename in fp.keys():
+            nodename = "%s_temp_%d" % (orig_nodename, tempcounter)
+            data.name(nodename)
+            tempcounter += 1
+data.hdf5_write(f"{filename_out}", directory=target_directory)
 print("\n*** FILE SAVED IN TARGET DIRECTORY ***\n")
-print(("Name of saved data", data.name()))
+print(
+    "saved data to (node, file, exp_type):",
+    data.name(),
+    filename_out,
+    my_exp_type,
+)
 config_dict.write()
-fl.show()
+# }}}
```

Comment:
note that I"ve made several comments or changes on either CPMG or generic, and you're going to want to diff the two.